Project MUSE.js

```diff
@@ -1,182 +1,116 @@
+function getSearchResults(doc, checkOnly) {
+    var items = {};
+    var found = false;
+    var rows = ZU.xpath(doc, '//div[@class="single_result"]//h1/a[contains(@href, "/article/") or contains(@href, "/book/")]');
+    for (var i=0; i<rows.length; i++) {
+        var href = rows[i].href;
+        var title = ZU.trimInternal(rows[i].textContent);
+        if (!href || !title) continue;
+        if (checkOnly) return true;
+        found = true;
+        items[href] = title;
+    }
+    return found ? items : false;
+}
 
-	var searchRe = new RegExp("^https?://[^/]+/search/results|/search/save|/results(#.+)?$");
-	if(detectWeb(doc, url) == "multiple") {
-		var items = {};
-		var attachments = new Array();
-		var results = ZU.xpath(doc,
-			'//div[@class="article"]/h4/a|//div[@class="result_info"]/h1/a');
-		if (results.length){
-			for(var i=0, n=results.length; i<n; i++) {
-				items[results[i].href] = results[i].textContent;
-			}
-		} else {
-			// Some journals have old-style TOCs for back issues
-			// Ex. http://muse.jhu.edu/journals/eighteenth-century_studies/toc/ecs33.4.html
-			var articles = doc.evaluate('//ul', doc, null, XPathResult.ANY_TYPE, null);
-			var article;
-			while (article = articles.iterateNext()) {
-				var link = doc.evaluate('./li/a[contains(@href,".html")]', article, null, XPathResult.ANY_TYPE, null).iterateNext();
-				var title = doc.evaluate('./li/i', article, null, XPathResult.ANY_TYPE, null).iterateNext();
-				if(link && link.href && title && title.textContent) {
-					items[link.href] = title.textContent;
-				}
-			}
-		}
 
-		Zotero.selectItems(items, function(items) {
-			if(!items) {
-				return true;
-			}	
-			var i;
-			var urls = [];
-			for (i in items) {urls.push(i);};
-			Zotero.Utilities.processDocuments(urls, scrapeOne, function() {Zotero.done();}, null);		
-		});
-	} else scrapeOne(doc);
-	Zotero.wait();
+function doWeb(doc, url) {
+    if (detectWeb(doc, url) == "multiple") {
+        Zotero.selectItems(getSearchResults(doc, false), function (items) {
+            if (!items) {
+                return true;
+            }
+            var articles = [];
+            for (var i in items) {
+                articles.push(i);
+            }
+            ZU.processDocuments(articles, scrape);
+        });
+    } else {
+        scrape(doc, url);
+    }
 }
 
-// Given an article page, get the RIS and open it
-function scrapeOne(doc) {
-	var url = doc.location.href;
-	var hostRe = new RegExp("^(https?://[^/]+)/");
-		var m = hostRe.exec(url);
-		var host = m[1];
+function scrape(doc, url) {
+	var abstract = ZU.xpathText(doc, '//div[@class="abstract"]');
+	if (!abstract) abstract = ZU.xpathText(doc, '//div[@class="description"][1]/text()');
+	var translator = Zotero.loadTranslator('web');
+    // Embedded Metadata
+    translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+    translator.setDocument(doc);
+    translator.setHandler('itemDone', function (obj, item) {
```

Comment:
The indent seems on these 4 lines wrong. Maybe check also above that you are using tabs and not spaces for the indent (when I copy and paste the code block this goes usually wrong).